packages/devtools_app/lib/src/shared/server/_analytics_api.dart

```diff
@@ -56,6 +56,32 @@ Future<bool> setAnalyticsEnabled([bool value = true]) async {
   return false;
 }
 
+/// Fetch the consent message for package:unified_analytics.
+Future<String?> getConsentMessage() async {
+  String? consentMessage;
+  if (isDevToolsServerAvailable) {
+    final resp = await request(apiGetConsentMessage);
+    if (resp?.statusOk ?? false) {
+      consentMessage = resp!.body;
+    }
+  }
+
+  return consentMessage;
+}
+
+/// Confirm with package:unified_analytics that the consent message
+/// has been shown to the user.
+Future<bool> confirmConsentMessageShown() async {
+  if (isDevToolsServerAvailable) {
+    final resp = await request(apiSetConsentMessageAsShown);
```

Comment:
I'm a bit confused by the contrasting api name and this method name. Are we setting a value or getting a value? 

Perhaps instead of `apiSetConsentMessageAsShown`, we could use `apiAnalyticsConsentMessageShown` and if that api call has an argument setting the value, then it is a set call, and if it does not, then it is a get call. This is similar to what is done for several boolean service extensions throughout the Flutter framework.

Reply:
or maybe I'm misunderstanding this - is there a need for getting the value of consent message shown? Or only setting?

Reply:
> is there a need for getting the value of consent message shown? Or only setting?

No, we don't need to get any value, this was a good catch because I made stub version of this function return `Future<void>`. I can swap out the web version of the api to be the same